src/werkzeug/routing.py

```diff
@@ -1663,6 +1695,9 @@ def __init__(
         self.path_info = to_unicode(path_info)
         self.default_method = to_unicode(default_method)
         self.query_args = query_args
+        self.websocket = websocket
+        if self.websocket:
+            self.url_scheme = "ws" if self.url_scheme == "http" else "wss"
```

Comment:
`url_scheme` is a parameter to `__init__`. Is there a reason this doesn't go the other way and say `self.websocket = url_scheme in {"ws", "wss"}`?

Reply:
Yea, this is nice, I've updated.